pkg/config/file.go

```diff
@@ -33,42 +36,80 @@ import (
 var _ Loader = (*FileLoader)(nil)
 
 type FileLoader struct {
-	path     string
+	config   LoaderConfig
 	cRuntime chan<- runtime.Config
 	done     chan struct{}
+	fsys     fs.FS
 }
 
 func NewFileLoader(cfg *Config, cRuntime chan<- runtime.Config) *FileLoader {
 	return &FileLoader{
-		path:     cfg.Loader.File.Path,
+		config:   cfg.Loader,
 		cRuntime: cRuntime,
+		done:     make(chan struct{}, 1),
+		fsys:     os.DirFS(filepath.Dir(cfg.Loader.File.Path)),
 	}
 }
 
+// Run gets the runtime configuration from the local file.
+// The config will be loaded periodically defined by the loader interval configuration.
+// Returns an error if the loader is shutdown or the context is done.
 func (f *FileLoader) Run(ctx context.Context) error {
+	ctx, cancel := logger.NewContextWithLogger(ctx)
+	defer cancel()
 	log := logger.FromContext(ctx)
-	log.Info("Reading config from file", "file", f.path)
-	// TODO refactor this to use fs.FS
-	b, err := os.ReadFile(f.path)
-	if err != nil {
-		log.Error("Failed to read config file", "path", f.path, "error", err)
-		return fmt.Errorf("failed to read config file: %w", err)
+	tick := time.NewTicker(f.config.Interval)
+	defer tick.Stop()
+
+	for {
+		select {
+		case <-f.done:
+			log.Info("File Loader terminated")
+			return nil
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-tick.C:
+			runtimeCfg, err := f.getRuntimeConfig(ctx)
+			if err != nil {
+				log.Warn("Could not get local runtime configuration", "error", err)
+				tick.Reset(f.config.Interval)
+				continue
+			}
+
+			log.Info("Successfully got local runtime configuration")
+			f.cRuntime <- runtimeCfg
+			tick.Reset(f.config.Interval)
+		}
 	}
+}
 
+// getRuntimeConfig gets the local runtime configuration from the specified file.
+func (f *FileLoader) getRuntimeConfig(ctx context.Context) (runtime.Config, error) {
+	log := logger.FromContext(ctx)
 	var cfg runtime.Config
 
+	file, err := f.fsys.Open(filepath.Base(f.config.File.Path))
+	if err != nil {
+		log.Error("Failed to open config file", "path", f.config.File.Path, "error", err)
+		return cfg, fmt.Errorf("failed to open config file: %w", err)
+	}
+	defer file.Close()
```

Comment:
Please handle error.

Reply:
Please look at it again. I've done some weird magic to still return an error if the file cannot be closed so that the user immediately notices that something is going (very) wrong.